packages/synapse-react-client/src/components/WikiMarkdownEditor/WikiMarkdownEditor.test.tsx

```diff
@@ -0,0 +1,442 @@
+  UNSAVED_CHANGES,
+  WikiMarkdownEditor,
+  WikiMarkdownEditorProps,
+} from './WikiMarkdownEditor'
+
+const onCancel = jest.fn()
+const onSave = jest.fn()
+
+const getRootWikiPageKeySpy = jest.spyOn(SynapseClient, 'getRootWikiPageKey')
+const getWikiPageSpy = jest.spyOn(SynapseClient, 'getWikiPage')
+const createWikiPageSpy = jest.spyOn(SynapseClient, 'createWikiPage')
+const updateWikiPageSpy = jest.spyOn(SynapseClient, 'updateWikiPage')
+
+const defaultNoRootProps: WikiMarkdownEditorProps = {
+  onSave,
+  onCancel,
+  ownerObjectId: '10000',
+  ownerObjectType: ObjectType.ENTITY,
+}
+const defaultExistingRootWikiPageProps: WikiMarkdownEditorProps = {
+  onSave,
+  onCancel,
+  ownerObjectId: mockEntityRootWikiPageKey.ownerObjectId,
+  ownerObjectType: mockEntityRootWikiPageKey.ownerObjectType,
+}
+const defaultExistingWikiPageProps: WikiMarkdownEditorProps = {
+  onSave,
+  onCancel,
+  ownerObjectId: mockEntityWikiPageKey.ownerObjectId,
+  ownerObjectType: mockEntityWikiPageKey.ownerObjectType,
+  wikiPageId: mockEntityWikiPageKey.wikiPageId,
+}
+
+const mockCreateRootWikiPage: CreateWikiPageInput['wikiPage'] = {
+  parentWikiId: undefined,
+  title: '',
+  markdown: '',
+  attachmentFileHandleIds: [],
+}
+
+function renderComponent(props: WikiMarkdownEditorProps) {
+  return render(<WikiMarkdownEditor {...props} />, {
+    wrapper: createWrapper(),
+  })
+}
+
+function getEditorDialog() {
+  return screen.getByRole('dialog', {
+    name: 'Edit Wiki Markdown',
+  })
+}
+
+function getSpinners() {
+  return document.getElementsByClassName('spinner')
+}
+
+function queryTitleField() {
+  return screen.queryByRole('textbox', { name: 'Title' })
+}
+
+function queryMarkdownField() {
+  return screen.queryByRole('textbox', { name: 'markdown' })
+}
+
+function setUp(props: WikiMarkdownEditorProps) {
+  const user = userEvent.setup()
+  const component = renderComponent(props)
+
+  const spinner = getSpinners()
+
+  return { component, user, spinner }
+}
+
+async function waitForWikiLoaded() {
+  await waitFor(() => expect(getSpinners()).toHaveLength(0))
+}
+
+async function waitForMarkdownSet(markdown: string) {
+  return await waitFor(() => {
+    const markdownField = queryMarkdownField()
+    expect(markdownField).not.toBeNull()
+    expect(markdownField).toHaveValue(markdown)
+    return markdownField!
+  })
+}
+
+async function waitForTitleSet(title: string) {
+  return await waitFor(() => {
+    const titleField = queryTitleField()
+    expect(titleField).not.toBeNull()
+    expect(titleField).toHaveValue(title)
+    return titleField!
+  })
+}
+
+describe('WikiMarkdownEditor', () => {
+  beforeEach(() => jest.clearAllMocks())
+  beforeAll(() => server.listen())
+  afterEach(() => server.resetHandlers())
+  afterAll(() => server.close())
+
+  test('creates a root WikiPage when there was not an existing root WikiPage', async () => {
+    const { spinner } = setUp(defaultNoRootProps)
+
+    expect(spinner).toHaveLength(1)
```

Comment:
I am surprised this consistently passes, I am always worried about checking something ephemeral in UI tests. If this works, that's great.